server/channels/api4/channel.go

```diff
@@ -2352,7 +2352,7 @@ func moveChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func getGroupMessageMembersCommonTeams(c *Context, w http.ResponseWriter, r *http.Request) {
+func getDirectOrGroupMessageMembersCommonTeams(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
Now that this function will be called much more often than before - on every DM.GM post operation; and the underlying SQL query being expensive and comprising of a subquery, it should be cached with some expiry.

We should use the automated load test command in this PR to see how much of a performance hit this causes (or maybe it doesn't) with the feature enabled vs disabled.